x-pack/plugin/src/test/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml

```diff
@@ -997,6 +1019,21 @@ setup:
 
   - match: { hits.total.value: 2 }
 
+  - do:
+      search:
+        index: .ml-stats-*
+        body:
+          size: 0
+          query:
+            bool:
+              filter:
+                ids:
+                  values:
+                    - "delete_foo_stats_1"
+                    - "delete_foo_stats_2"
+
+  - match: { hits.total.value: 2 }
```

Comment:
Oddly this returns 0 hits when run with security. I can't see any difference between the `.ml-stats` and the `.ml-state` indices. The same search is done for the latter just before and it works fine. I tried adding the header and that fixes it. But it's not needed when searching the state index. Any ideas @droberts195 ?

Reply:
There is some difference between security and no-security cases when it comes to hidden indices wildcard expansion.
Having said that, I don't understand the difference between `.ml-state*` and `.ml-stats-*` as both are hidden...

Reply:
I think you should give the `machine_learning_admin` role `read` and `view_index_metadata` privileges on the `.ml-stats*` pattern.

`machine_learning_admin` has read access to the `.ml-state*` pattern, so that is the difference.